### PR TITLE
Basic permissions

### PR DESCRIPTION
**Pull request by JavkhlanShirendev**

### CODE REVIEW

**Comment by JavkhlanShirendev** [comment]
Please review @batamar 
Can't set permissions on social integration mutations, ... i can't not find them. @batamar

File: src/data/permissions.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { ROLES } from './constants';
+
+export const PERMISSIONS = {
```
Comment (batamar) [inline src/data/permissions.js]: use Roles.ADMIN instead

File: src/data/permissions.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  ADMIN: 'admin',
+};
+
+export const checkPermission = (permission, user) => {
+  if (permission === PERMISSIONS.ADMIN && user.role === ROLES.CONTRIBUTOR) {
+    throw new Error('Permission required');
+  }
+};
+
+export const requireLogin = (cls, methodName, permissions) => {
```
Comment (batamar) [inline src/data/permissions.js]: Write function comment

File: src/data/permissions.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    for (let permission of permissions || []) {
+      checkPermission(permission, user);
+    }
+
+    return oldMethod(root, object2, { user });
+  };
+};
+
+export const moduleRequireLogin = (mdl, permissions) => {
```
Comment (batamar) [inline src/data/permissions.js]: Write function comment

**Pull request merged**